tools/cache/time-defined-cache_test.go

```diff
@@ -71,49 +72,41 @@ var _ = Describe("time defined cache", func() {
 		Expect(err).To(HaveOccurred())
 	})
 
-	It("should not allow two threads to set value in parallel", func() {
-		stopChannel := make(chan struct{})
-		defer close(stopChannel)
-		firstCallMadeChannel := make(chan struct{})
-
-		recalcFunctionCalls := int64(0)
+	It("when multiple go routines get a value only one is recalculating and others get the same cached value", func() {
+		firstCallBarrier := make(chan struct{})
```

Comment:
I was thinking it's necessary too, and had these in previous versions of this PR, but then I saw this from the official golang tour [1]:
> Another note: Channels aren't like files; you don't usually need to close them. Closing is only necessary when the receiver must be told there are no more values coming, such as to terminate a range loop.

If it is not necessary, it bring more damage than good because we both make and code less readable and educate to do something that's not needed.

[1] https://go.dev/tour/concurrency/4

Reply:
Learning something new! :)

Reply:
Yeah it surprised me as well :)
Good to know :smile: